lib/utils.pm

```diff
@@ -345,6 +345,7 @@ sub zypper_call {
     my $dumb_term        = $args{dumb_term};
 
     my $printer = $log ? "| tee /tmp/$log" : $dumb_term ? '| cat' : '';
+    die 'Exit code is from PIPESTATUS[0], not grep' if $command =~ /^((?!`).)*\| ?grep/;
```

Comment:
Would be cool if we can generalize this to cover more than just `grep` but I also do not know how this could be done so let's go for it as is :)

Reply:
@okurz I check for grep because it's usual way of checking output `$cmd | grep OK`
We can forbid all $commands which contain pipe, but it would probably break stuff.